workflow/controller/operator.go

```diff
@@ -873,7 +916,11 @@ func (woc *wfOperationCtx) podReconciliation(ctx context.Context) error {
 		if pod == nil {
 			return
 		}
+		if isAgentPod(pod) {
+			woc.reconcileAgentNode(pod)
```

Comment:
you should just ignore agent pods, we don't care about them at all, all comms is done via the taskset

Reply:
This code need if the Agent pod errored or failed during the creation/running time

Reply:
ah - so if agent pod errors, then workflow errors? that makes sense - a workflow pod failing would do the same